src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -20,18 +21,21 @@ public class InstanceController : Controller
     {
         private readonly IEnvironment _environment;
         private readonly IInstanceManager _instanceManager;
+        private readonly ILogger _logger;
 
-        public InstanceController(IEnvironment environment, IInstanceManager instanceManager)
+        public InstanceController(IEnvironment environment, IInstanceManager instanceManager, ILoggerFactory loggerFactory)
         {
             _environment = environment;
             _instanceManager = instanceManager;
+            _logger = loggerFactory.CreateLogger<InstanceController>();
         }
 
         [HttpPost]
         [Route("admin/instance/assign")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
         public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContext encryptedAssignmentContext)
         {
+            _logger.LogDebug($"Starting container assignment for host : {Request.Host}");
```

Comment:
Let's also log the context payload size (e.g. ContextLength=encryptedAssignmentContext?.EncryptedContext.Length). That would have told us that the payload was empty in the bug we saw?